cmd/ipfs/daemon.go

```diff
@@ -759,6 +764,43 @@ func rewriteMaddrToUseLocalhostIfItsAny(maddr ma.Multiaddr) ma.Multiaddr {
 	}
 }
 
+// printRessourceManagerLimit fetches the user-defined resource overrides, and then
+// displays the limits for MaxMemory and MaxFileDescriptors
+func printRessourceManagerLimit(bcfg *node.BuildCfg, cfg *config.Config) error {
+	enabled := cfg.Swarm.ResourceMgr.Enabled.WithDefault(true)
+	switch os.Getenv("LIBP2P_RCMGR") {
+	case "0", "false":
+		enabled = false
+	case "1", "true":
+		enabled = true
+	}
```

Comment:
https://github.com/ipfs/kubo/blob/e4908a01633582e205b26ab36769830fe8490c10/core/node/libp2p/rcmgr.go#L37-L43

It's weird that this code is duplicated, it's confusing and will make it anoying to change `LIBP2P_RCMGR`, I expect control flow (not call this function if disabled) or a field in one of the structs or an argument to be passed instead.

Reply:
it make sense, I've updated my commit, do you see a better way do this ?